src/Compilers/CSharp/Test/Symbol/Symbols/Source/MethodTests.cs

```diff
@@ -203,6 +204,84 @@ public partial class A {
         }
 
         [Fact]
+        public void PartialExtractSyntaxLocation_DeclBeforeDef()
+        {
+            var text =
+@"public partial class A {
+  partial void M();
+}
+public partial class A {
+  partial void M() {}
+}
+";
+            var comp = CreateStandardCompilation(text);
+            var global = comp.GlobalNamespace;
+            var a = global.GetTypeMembers("A", 0).Single();
+            var m = a.GetMembers("M");
+            Assert.Equal(1, m.Length);
+            var returnSyntax = ((MethodSymbol) m.First()).ExtractReturnTypeSyntax();
+            var newLineCount = Environment.NewLine.Length * 1;
+            Assert.Equal(returnSyntax.Location.SourceSpan, TextSpan.FromBounds(34 + newLineCount, 38 + newLineCount));
```

Comment:
Once you find the node, simply assert that the nodes are the same.

---
In reply to: [113774955](https://github.com/dotnet/roslyn/pull/18472#discussion_r113774955) [](ancestors = 113774955)